client/client_test.go

```diff
@@ -3311,6 +3469,173 @@ func testLazyImagePush(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 }
 
+func testZstdLocalCacheExport(t *testing.T, sb integration.Sandbox) {
+	c, err := New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	busybox := llb.Image("busybox:latest")
+	cmd := `sh -e -c "echo -n zstd > data"`
+
+	st := llb.Scratch()
+	st = busybox.Run(llb.Shlex(cmd), llb.Dir("/wd")).AddMount("/wd", st)
+
+	def, err := st.Marshal(sb.Context())
+	require.NoError(t, err)
+
+	destDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(destDir)
+
+	destOutDir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(destOutDir)
+
+	_, err = c.Solve(sb.Context(), def, SolveOpt{
+		Exports: []ExportEntry{
+			{
+				Type:      ExporterLocal,
+				OutputDir: destOutDir,
+			},
+		},
+		// compression option should work even with inline cache exports
+		CacheExports: []CacheOptionsEntry{
+			{
+				Type: "local",
+				Attrs: map[string]string{
+					"dest":        destDir,
+					"compression": "zstd",
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+
+	var index ocispecs.Index
+	dt, err := ioutil.ReadFile(filepath.Join(destDir, "index.json"))
+	require.NoError(t, err)
+	err = json.Unmarshal(dt, &index)
+	require.NoError(t, err)
+
+	var layerIndex ocispecs.Index
+	dt, err = ioutil.ReadFile(filepath.Join(destDir, "blobs/sha256/"+index.Manifests[0].Digest.Hex()))
+	require.NoError(t, err)
+	err = json.Unmarshal(dt, &layerIndex)
+	require.NoError(t, err)
+
+	lastLayer := layerIndex.Manifests[len(layerIndex.Manifests)-2]
+	require.Equal(t, ocispecs.MediaTypeImageLayer+"+zstd", lastLayer.MediaType)
+
+	zstdLayerDigest := lastLayer.Digest.Hex()
+	dt, err = ioutil.ReadFile(filepath.Join(destDir, "blobs/sha256/"+zstdLayerDigest))
+	require.NoError(t, err)
+	require.Equal(t, dt[:4], []byte{0x28, 0xb5, 0x2f, 0xfd})
+}
+
+func testUncompressedLocalCacheImportExport(t *testing.T, sb integration.Sandbox) {
+	skipDockerd(t, sb)
+	dir, err := ioutil.TempDir("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+	im := CacheOptionsEntry{
+		Type: "local",
+		Attrs: map[string]string{
+			"src": dir,
+		},
+	}
+	ex := CacheOptionsEntry{
+		Type: "local",
+		Attrs: map[string]string{
+			"dest":              dir,
+			"compression":       "uncompressed",
+			"force-compression": "true",
+		},
+	}
+	testBasicCacheImportExport(t, sb, []CacheOptionsEntry{im}, []CacheOptionsEntry{ex})
+}
+
+func testUncompressedRegistryCacheImportExport(t *testing.T, sb integration.Sandbox) {
```

Comment:
@ktock This new test failed in the merge commit https://github.com/moby/buildkit/runs/5425504420?check_suite_focus=true